src/Modules/UngradedModules/SubQuoHom.jl

```diff
@@ -329,14 +329,33 @@ function is_welldefined(H::SubQuoHom)
   return true
 end
 
-function (==)(f::ModuleFPHom, g::ModuleFPHom)
+# No ring map
+function (==)(f::ModuleFPHom{D, C, Nothing}, g::ModuleFPHom{D, C, Nothing}) where {D, C}
   domain(f) === domain(g) || return false
   codomain(f) === codomain(g) || return false
-  M = domain(f)
-  for v in gens(M)
-    f(v) == g(v) || return false
+  return all(f(v) == g(v) for v in gens(domain(f)))
+end
+
+# With ring map
+function (==)(f::ModuleFPHom, g::ModuleFPHom)
+  if isnothing(ring_map(f))
+    isnothing(ring_map(g)) || is_trivial(ring_map(g)) || return false
   end
-  return true
+  if isnothing(ring_map(g))
+    isnothing(ring_map(f)) || is_trivial(ring_map(f)) || return false
+  end
+  # TODO: Catch the case where == defaults to === and this returns false, 
+  # but the ring_maps are identical in the mathematical sense nevertheless.
```

Comment:
I am a bit puzzled with the problem we're facing here. I would like to throw an error in case comparison of maps is not implemented. But given the freedom that we gave to the maps of rings, it seems unavoidable that for numerous combinations `==` will just default to `===` and then return `false`, even though the maps might be mathematically the same or not comparable. 

But this must have popped up elsewhere, too. How is it solved, for instance, for comparison of morphisms of polynomial rings?

Reply:
> I guess you implemented this here

Indeed, I did. And I didn't do it properly and had to correct it here in the latest version.